contracts/DAORegistry.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./ContributorRegistry.sol";
+import "./RoundManager.sol";
+import "./Mock/MockEventStream.sol";
+import "./Mock/MockPaymentStream.sol";
+
+/* This is the Rhada parent contract. Here is the following flow:
+ * DAOs will register to join the Rhada protocol and will list the following:
+ *      - Address of the contract/user that will interact with the Rhada Protocol
+ *      - Funds/Fund type: funds can come later but addresses of the DAO's ERC20 token
+ * Upon registering,
+ *      - Will deploy a new ContributorRegistry contract specific to that DAO
+ *              - Manages devs, voting process for who is a contributor
+ *              - Need to launch here, don't need in struct. RoundManager uses its data, not this contract
+ *      - Will deploy a new RoundVoting contract specific to that DAO
+ *              - Manages rounds and voting on specific users in a round. Used to define how salary gets split
+ *              - Possible (and probably necessary) feature: rewards DAO members for participating
+ *      - Will deploy a PaymentStream contract specific to that DAO
+ *              - Controls the setting up of payments
+ *      - Will deploy an EventStream contract specific to that DAO
+ *              - Controls monitoring of no idea
+ * DAO parent address will map to address of each one
+ * Contracts compiles data from all of them
+ *      - ContributorRegistry
+ *          - Don't know what to use from this
+ *      - RoundVoting
+ *          - Needs votes + users (will use to split salary among them all)
+ *      - PaymentStream
+ *          - Will start payment stream from this contract
+ * Also managed here:
+ *      - Weighted voting (DAO -> weighted voting. Can act as a user registry per DAO)
+ *      - Lists all contributors, payments streams, and other important data
+*/
+
+
+contract DAORegistry is Ownable {
+    /* ============ Datatypes ============ */
+    struct DAO {
+        ContributorRegistry contributorRegistry; // Probably don't need b/c used in roundManager, RM has sufficient checks. Still need to deploy though
```

Comment:
Generally when we manage the Contributor Registry here we will have to change the way we currently integrate it in the frontend.